main.go

```diff
@@ -77,6 +58,50 @@ func createNewArticle(w http.ResponseWriter, r *http.Request) {
 	// fmt.Fprintf(w, "%+v", string(reqBody))
 }
 
+func deleteArticle(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	key := vars["id"]
+	for index, article := range Articles {
+		if article.Id == key {
+			Articles = append(Articles[:index], Articles[index+1:]...)
+		}
+	}
+}
+
+func updateArticle(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	key := vars["id"]
+	for index, article := range Articles {
+		if article.Id == key {
+			reqBody, _ := ioutil.ReadAll(r.Body)
+			var updatedArticle Article
+			json.Unmarshal(reqBody, &updatedArticle)
+			// Is there an easy way to spread prev values and override only some properties?
+			// combinedUpdatedArticle := {...Articles[index], ...updatedArticle}
+			Articles[index] = updatedArticle
+			json.NewEncoder(w).Encode(updatedArticle)
+		}
+		// how to return error if id not found?
+	}
+}
+
+type Email struct {
+	Subject string `json:"subject"`
+	Body    string `json:"body"`
+}
+
+// type Dadjoke struct {
+// 	Id     string `json:"id"`
+// 	Joke   string `json:"joke"`
+// 	Status int    `json:"status"`
+// }
+
+type Dadjoke struct {
+	Id     string
+	Joke   string
+	Status int
+}
+
```

Comment:
Is this the proper way to handle the errors in getDadJoke below? 🤔

Reply:
ya so you would want to return any error you got from this function so that you can bubble it up and potentially tell the user. so instead of just printing errors make them `return "", err` and set your return type to `(string, error)` you can return multiple things.

Also you can re use the `err` variable instead of doing `anothererr`. 
and your final return would look like `return dadJoke.Joke, nil` since there was no error.

Reply:
then when you call getDadJoke you would need to handle the err there

Reply:
sendFunnyEmail is your handler for the endpoint. It has access to the request and response. this is where you can write errors much like your success response 
`response.WriteHeader(status)`
`_ = json.NewEncoder(response).Encode(err.Error)`